File: wiz-shop-java/src/main/java/com/example/wizshopjava/api/member/AuthController.java
package com.example.wizshopjava.api.member;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.wizshopjava.api.member.dto.MemberLoginRequest;
import com.example.wizshopjava.api.member.dto.MemberRegisterRequest;

import lombok.RequiredArgsConstructor;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {

	private final MemberService memberService;

	@PostMapping("/register")
	public ResponseEntity<Void> register(@RequestBody MemberRegisterRequest req) {
		memberService.register(req);
		return ResponseEntity.ok().build();
	}

	@PostMapping("/login")
	public ResponseEntity<String> login(@RequestBody MemberLoginRequest req) {
		String accessToken = memberService.login(req);
		return ResponseEntity.ok(accessToken);
	}
}

File: wiz-shop/src/main/kotlin/com/example/wizshop/api/member/AuthController.kt
package com.example.wizshop.api.member

import com.example.wizshop.api.member.dto.MemberLoginRequest
import com.example.wizshop.api.member.dto.MemberRegisterRequest
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/auth")
class AuthController(
    private val memberService: MemberService
) {

    @PostMapping("/register")
    fun register(@RequestBody req: MemberRegisterRequest): ResponseEntity<Unit> {
        return memberService.register(req)
            .let { ResponseEntity.ok(it) }
    }

    @PostMapping("/login")
    fun login(@RequestBody req: MemberLoginRequest): ResponseEntity<String> {
        return memberService.login(req)
            .let { ResponseEntity.ok(it) }
    }
}